src/VisualStudio/Core/Def/Snippets/ISnippetExpansionClientFactory.cs

```diff
@@ -0,0 +1,14 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.CodeAnalysis.Host;
+using Microsoft.VisualStudio.Text;
+using Microsoft.VisualStudio.Text.Editor;
+
+namespace Microsoft.VisualStudio.LanguageServices.Implementation.Snippets;
+
+internal interface ISnippetExpansionClientFactory : ILanguageService
+{
+    AbstractSnippetExpansionClient GetSnippetExpansionClient(ITextView textView, ITextBuffer subjectBuffer);
```

Comment:
nit: for the future would prefer if this exposed a particular interface.  I don't love interfaces exposing abstract-reified types like this.  But it's def not blocking, and i don't mind this corner of the LS working this way.

Reply:
➡️ Left this one as-is